version: '3.9'
services:
  django_db:
    container_name: django_db
    image: postgres:15.2-bullseye
    volumes:
      - ./postgres_data:/var/lib/postgresql/data
    env_file:
      - ./.env.db
    ports:
      - "5432:5432"
    healthcheck:
      test: [ "CMD-SHELL", "pg_isready -U ipankratov" ]
      interval: 5s
      timeout: 5s
      retries: 5
      
  django:
    build:
      context: ./web
      dockerfile: Dockerfile
    container_name: django
    command:
      sh ./dev.start.sh
    volumes:
      - ./web/backend:/usr/backend
    env_file:
      - ./web/.env
    depends_on:
      - django_db
    tty: true
    ports:
      - "8000:8000"
    links:
      - django_db:django_db
volumes:
  postgres_data:
